builtin/providers/kubernetes/resource_kubernetes_resource_quota.go

```diff
@@ -0,0 +1,211 @@
+package kubernetes
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	pkgApi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	api "k8s.io/kubernetes/pkg/api/v1"
+	kubernetes "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+)
+
+func resourceKubernetesResourceQuota() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKubernetesResourceQuotaCreate,
+		Read:   resourceKubernetesResourceQuotaRead,
+		Exists: resourceKubernetesResourceQuotaExists,
+		Update: resourceKubernetesResourceQuotaUpdate,
+		Delete: resourceKubernetesResourceQuotaDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("resource quota", true),
+			"spec": {
+				Type:        schema.TypeList,
+				Description: "Spec defines the desired quota. http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status",
+				Optional:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"hard": {
+							Type:         schema.TypeMap,
+							Description:  "The set of desired hard limits for each named resource. More info: http://releases.k8s.io/HEAD/docs/design/admission_control_resource_quota.md#admissioncontrol-plugin-resourcequota",
+							Optional:     true,
+							Elem:         schema.TypeString,
+							ValidateFunc: validateResourceList,
+						},
+						"scopes": {
+							Type:        schema.TypeSet,
+							Description: "A collection of filters that must match each object tracked by a quota. If not specified, the quota matches all objects.",
+							Optional:    true,
+							ForceNew:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Set:         schema.HashString,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesResourceQuotaCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*kubernetes.Clientset)
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
```

Comment:
That's merely because `expandResourceQuoteSpec` has to parse "resource list" and that may error out, `expandMetadata` doesn't have to parse anything like that.
https://github.com/hashicorp/terraform/pull/13914/files#diff-886d9a6a55a7569117aaf167990fed25R223